### ISSUE TITLE
De-hardcode URLs in code

### ISSUE THREAD

**Issue by slinkp** [openplans/openblock#69]
There's a lot of stuff like this in ebpub.db.models:

```
    def item_url(self):
        return '/%s/by-date/%s/%s/%s/%s/' % (self.schema.slug, self.item_date.year, self.item_date.month, self.item_date.day, self.id)
```

That can/should be replaced by suitable call to reverse().

**Comment by slinkp** [comment]
There is also metro-aware code in ebpub/ebpub/utils/urlresolvers.py
(currently only imported in ebpub/ebpub/urls.py)

and @permalink might be useful: http://docs.djangoproject.com/en/dev/ref/models/instances/#django.db.models.permalink

**Comment by slinkp** [comment]
Note some of the common views (place_detail_timeline, place_detail_overview) can't be fed to reverse() because they require both a varying number of positional args AND the kwarg 'place_type' , and reverse() does not allow passing both kwargs and positional args.

Those views arguably do too much anyway, as they handle both Blocks and Locations in single-city and multi-city situations, relying on their own URL parsing to decide what to do.  Maybe it would be possible to break that down into a few URL patterns hooked up to a few view functions that take either only kwargs or only positional args.  And place_type could be baked in, rather than an arg.
Common functionality could still be factored into other functions and/or template(s).
There'd be more view functions to understand, but they'd be smaller and simpler -- and the URL config would be more explicit -- and reverse() would work.

**Comment by slinkp** [comment]
(In [9c02185a376bf64093f638ae20786259118292e5]) Change many generated URLs to use urlresolvers.reverse(). Closes #69
- Remove ebpub.utils.urlresolvers, replaced by a simple conditional in ebpub.urls.
- Name a lot more of the URLs in ebpub.urls.

**Comment by slinkp** [comment]
Ticket imported from Trac:
 http://developer.openblockproject.org/ticket/69
Reported by: slinkp
